components/Popover.php

```diff
@@ -0,0 +1,518 @@
+			'icon'           => $this->popover_menu_item_icon,
+			'icon_alignment' => $this->popover_menu_item_icon_alignment,
+			'attr'           => $args['attr'] ?? array(),
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Render popover header element.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string
+	 */
+	protected function render_header(): string {
+		if ( empty( $this->popover_title ) ) {
+			return '';
+		}
+
+		$title = sprintf( '<h3 class="tutor-popover-title">%s</h3>', esc_attr( $this->popover_title ) );
+
+		$close_button = sprintf(
+			'<button @click="hide()" class="tutor-popover-close">
+				%s
+			</button>',
+			tutor_utils()->get_svg_icon( Icon::CROSS, 14, 14 )
+		);
+
+		if ( $this->show_close_button ) {
+			return sprintf(
+				'<div class="tutor-popover-header">
+					%s
+					%s
+				</div>
+			',
+				$title,
+				$close_button
+			);
+		}
+
+		return sprintf(
+			'<div class="tutor-popover-header">
+					%s
+				</div>
+			',
+			$title,
+		);
+	}
+
+	/**
+	 * Render popover body component.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string
+	 */
+	protected function render_body(): string {
+		if ( empty( $this->popover_body ) ) {
+			return '';
+		}
+
+		$body = $this->esc( $this->popover_body, $this->popover_body_esc );
+
+		return sprintf(
+			'<div class="tutor-popover-body">
+				%s
+			</div>',
+			$body
+		);
+	}
+
+	/**
+	 * Render footer component for popover.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string
+	 */
+	protected function render_footer(): string {
+		$footer_buttons   = '';
+		$footer_alignment = in_array( $this->popover_footer_alignment, array( self::LEFT, self::CENTER ), true ) ? $this->popover_footer_alignment : '';
+
+		if ( ! tutor_utils()->count( $this->popover_footer_buttons ) ) {
+			return $footer_buttons;
+		}
+
+		$alignment = self::LEFT === $footer_alignment ? 'tutor-justify-start' : 'tutor-justify-center';
+
+		foreach ( $this->popover_footer_buttons as $button ) {
+			$footer_buttons .= $button;
+		}
+
+		if ( ! empty( $footer_alignment ) ) {
+			return sprintf(
+				'<div class="tutor-popover-footer %s">
+					%s
+				</div>',
+				$alignment,
+				$footer_buttons
+			);
+		}
+
+		return sprintf(
+			'<div class="tutor-popover-footer">
+				%s
+			</div>',
+			$footer_buttons
+		);
+	}
+
+	/**
+	 * Handle closing popover on outside click.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param boolean $is_closeable whether allow to close from outside click.
+	 *
+	 * @return self
+	 */
+	public function dismissible( bool $is_closeable ): self {
+		$this->popover_close_outside = $is_closeable;
+		return $this;
+	}
+
+	/**
+	 * Render Menu for popover.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string
+	 */
+	protected function render_menu(): string {
+		$menu_items = '';
+		if ( ! tutor_utils()->count( $this->popover_menu_items ) ) {
+			return $menu_items;
+		}
+
+		foreach ( $this->popover_menu_items as $item ) {
+			$tag            = isset( $item['tag'] ) ? $item['tag'] : 'div';
```

Comment:
We can use the shorthand:
Longhand: `isset( $item['tag'] ) ? $item['tag'] : 'div';`
Shorthand: `$item['tag']  ??  'div';`